http-client/src/main/java/io/airlift/http/client/HttpClient.java

```diff
@@ -25,6 +25,10 @@ public interface HttpClient
     <T, E extends Exception> T execute(Request request, ResponseHandler<T, E> responseHandler)
             throws E;
 
+    /**
+     * Executes request asynchronously running responseHandler on the client's http thread pool.
+     */
+    @Deprecated
```

Comment:
add `@deprecated` javadoc tag explaining why it's deprecated and what to use instead.

This generally feels better if it's part of `Add HttpClient.executeAsync that accepts Executor` commit.

Reply:
Added `@deprecated {@link }`